web.config

```diff
@@ -228,41 +228,66 @@
                   <action type="Redirect" url="/keyboards/h/{R:1}{R:2}" redirectType="Permanent" />
                 </rule>
 
-                <!-- Masking of PHP filename suffix -->
-                <rule name="Ignore .php extension">
+                <!-- PHP and Markdown rewriting -->
+
+                <rule name="Remove index or index.php and redirect" stopProcessing="true">
+                  <match url="^((.+)/)?index(\.php)?$" />
+                  <action type="Redirect" url="{R:1}" />
+                </rule>
+
+                <rule name="Remove .php extension and redirect" stopProcessing="true">
                   <match url="^(.+)\.php$" />
                   <conditions logicalGrouping="MatchAll" trackAllCaptures="false">
                     <add matchType="IsFile" />
                   </conditions>
                   <action type="Redirect" url="{R:1}" />
                 </rule>
 
-                <rule name="Redirect index or index.php" stopProcessing="true">
-                  <match url="((.+)/)?index(\.php)?$" />
-                  <action type="Redirect" url="{R:1}" />
+                <rule name="Redirect folder without / to include /" stopProcessing="true">
+                  <match url="^(.+[^/])$" />
+                  <conditions logicalGrouping="MatchAll" trackAllCaptures="false">
+                    <add matchType="IsDirectory" />
+                    <add matchType="IsFile" negate="true" input="{DOCUMENT_ROOT}/{R:1}.php" />
+                    <add matchType="IsFile" negate="true" input="{DOCUMENT_ROOT}/{R:1}.md" />
+                  </conditions>
+                  <action type="Redirect" url="{R:1}/" />
                 </rule>
 
-                <rule name="Ignore folders" stopProcessing="true">
-                  <match url="^(.+)/$" />
-                  <conditions logicalGrouping="MatchAll" trackAllCaptures="false" />
-                  <action type="Rewrite" url="{R:1}/index.php" />
-                </rule>
+                <!-- PHP rewriting -->
 
-                <rule name="Is a folder" stopProcessing="true">
+                <rule name="Rewrite file to file.md" stopProcessing="false">
                   <match url="^(.+)$" />
                   <conditions logicalGrouping="MatchAll" trackAllCaptures="false">
-                    <add matchType="IsDirectory" />
+                    <add matchType="IsFile" input="{DOCUMENT_ROOT}/{R:1}.md" />
                   </conditions>
-                  <action type="Redirect" url="{R:1}/" />
+                  <action type="Rewrite" url="/_common/mdhost.php?file={R:1}.md" />
                 </rule>
 
-                <rule name="Add .php extension">
+                <rule name="Rewrite file to file.php" stopProcessing="false">
```

Comment:
```suggestion
                <rule name="Rewrite file to file.php" stopProcessing="true">
```

Reply:
I've made these updates to web.config.

I originally got them from help.keyman.com so I guess we'll need to fix there too

Reply:
> I originally got them from help.keyman.com so I guess we'll need to fix there too

Low priority I think.